src/main/java/com/algorand/algosdk/account/Account.java

```diff
@@ -255,12 +259,17 @@ static public void setFeeByFeePerByte(Transaction tx, BigInteger suggestedFeePer
      * @return an estimated byte size for the transaction.
      */
     public static BigInteger estimatedEncodedSize(Transaction tx) throws NoSuchAlgorithmException {
-        Account acc = new Account();
-        try {
-            return BigInteger.valueOf(Encoder.encodeToMsgPack(acc.signTransaction(tx)).length);
-        } catch (IOException e) {
-            throw new RuntimeException("unexpected behavior", e);
-        }
+    	try {
+    		long length = Encoder.encodeToMsgPack(
+    				new SignedTransaction(
+    						tx, 
+    						new Account().rawSignBytes(
+    								Arrays.copyOf(tx.bytesToSign(), tx.bytesToSign().length)),
+    						tx.txID())).length;
+    		return BigInteger.valueOf(length);
+    	} catch (IOException e) {
+    		throw new RuntimeException("unexpected behavior", e);
+    	}
```

Comment:
mixed tabs/spaces here

Reply:
Could you explain what is happening here? I suppose this is to account for the AuthAddr being set in the signed transaction?

Is this to match with the golden, and do the legacy tests need to be updated?

Was something like this needed in the other SDKs?

Reply:
The transaction is signed with a random account to get the size of the signed transaction. SignTransaction does the signing, but also sets authAddr which is not desired here.
Legacy tests do not need to be updated. 
This is the consequence of having the Account sign a Transaction, and being aware of authAddr.

Reply:
Ok, if the existing tests still pass that should be enough. I think our program size estimates aren't perfect in general but that is outside the scope of this feature.